Cargo.toml

```diff
@@ -52,6 +52,7 @@ num-complex = "0.4"
 num-traits = "0.2"
 probability = "0.20"
 indenter = "0.3"
+itertools = "0.10.5"
```

Comment:
Looking closer, I don't think this dependency is actually needed... I'll clarify in other comments.

Reply:
The other comments explain stdlib alternatives to `collect_vec` and `sorted` so that this dependency can be dropped.

Reply:
After merging your suggestions that removed uses of itertools, I removed this dependency, thanks Stefan!

Reply:
Whoops, looks like the usage in the module was removed but this line is still pulling it into the dependencies even though nothing uses it. This can be removed now too:

```suggestion
```